lib/entities/app-action.ts

```diff
@@ -66,6 +66,14 @@ type BaseAppActionProps = AppActionCategory & {
    * Human readable description of the action
    */
   description?: string
+  /**
+   * Optional JSON Schema describing the request payload shape
+   */
+  parametersSchema?: object
+  /**
+   * Optional JSON Schema describing the result shape
+   */
+  resultSchema?: object
```

Comment:
`object` includes arrays, functions, dates, regexes, so let's scope this down to more of what a JSONSchema would look like
```suggestion
  parametersSchema?: Record<string, unknown>
  /**
   * Optional JSON Schema describing the result shape
   */
  resultSchema?: Record<string, unknown>
```